src/main/java/school/lemon/changerequest/java/generics/Generator/Printer.java

```diff
@@ -0,0 +1,20 @@
+package school.lemon.changerequest.java.generics.Generator;
+
+public class Printer<T extends Number> {
+    NumberGenerator<T> numberGenerator;
+    SumCalculator<T> sum;
+
+    public Printer(NumberGenerator<T> numberGenerator) {
+        this.numberGenerator = numberGenerator;
+    }
+
+    public void generateAndPrint() {
+        T[] arrayNumbers = numberGenerator.generateNumbers();
+        if (numberGenerator instanceof IntegerGenerator) {
+            sum = (SumCalculator<T>) new IntegerSumCalculator();
```

Comment:
1. It's better to move creation of SumCalculator to constructor.
2. It's mot flexible, because there is no way to add new type of number generator.
I'd prefer to create additional method in NumberGenerator that will return specific SumCalculator.